resolver/resolver.go

```diff
@@ -311,15 +287,9 @@ func (res *Resolver) HandleNonMesos(w dns.ResponseWriter, r *dns.Msg) {
 		// set refused
 		m.SetRcode(r, 5)
 	} else {
-
-		proto := "udp"
-		if _, ok := w.RemoteAddr().(*net.TCPAddr); ok {
```

Comment:
i'd feel much better if we left `proto` as part of the interface for now. we can always yank it out in a later PR once we've verified that there no users depending on this. this PR seems like the wrong place to embed a behavioral change like this.